toolkit/cli/commands/src/registration_signatures.rs

```diff
@@ -79,6 +172,25 @@ impl RegisterValidatorMessage {
 		}
 	}
 
+	/// Generates dual signatures and prepares the complete registration output.
+	///
+	/// This method performs the cryptographic signing operations required for
+	/// validator registration, generating both Ed25519 and ECDSA signatures
+	/// over this message structure.
+	///
+	/// # Arguments
+	/// * `mainchain_key` - Ed25519 signing key for the Cardano stake pool
+	/// * `sidechain_key` - ECDSA signing key for the Partner Chain validator
+	///
+	/// # Returns
+	/// A `RegistrationCmdOutput` containing all signatures and public keys
```

Comment:
It is not true if "all" is meant as a quantifier of "public keys"

Reply:
Also `RegistrationCmdOutput` could be referenced by type here and there is no need to repeat